src/bdmsupport.c

```diff
@@ -743,10 +794,35 @@ int bdmUpdateDeviceData(item_list_t *itemList)
             pDeviceData->bdmDeviceType = BDM_TYPE_ILINK;
         else if (!strcmp(pDeviceData->bdmDriver, "sdc") && strlen(pDeviceData->bdmDriver) == 3)
             pDeviceData->bdmDeviceType = BDM_TYPE_SDC;
-        else
+        else if (!strcmp(pDeviceData->bdmDriver, "ata") && strlen(pDeviceData->bdmDriver) == 3) {
+            pDeviceData->bdmDeviceType = BDM_TYPE_ATA;
+            itemList->flags = MODE_FLAG_COMPAT_DMA;
+        } else
             pDeviceData->bdmDeviceType = BDM_TYPE_UNKNOWN;
 
-        LOG("Mass device: %d (%d) %s -> %s\n", itemList->mode, pDeviceData->massDeviceIndex, pDeviceData->bdmPrefix, pDeviceData->bdmDriver);
+        // If the device is backed by the ATA driver then get the supported LBA size for the drive.
+        if (pDeviceData->bdmDeviceType == BDM_TYPE_ATA) {
+            // If atad is loaded then xhdd is also loaded, query the hdd to see if it supports LBA48 or not.
```

Comment:
Isn't all of that info already loaded into atad_devinfo_t structure during atad driver initialization? not quite sure, maybe this data is private and it is easier to calculate it one more time.

Reply:
looks like we still need to call the cmds here to retrieve it